arrow-data/src/transform/mod.rs

```diff
@@ -718,6 +720,28 @@ impl<'a> MutableArrayData<'a> {
         self.data.len += len;
     }
 
+    /// Extends the in progress array with the same value from the input arrays
+    ///
+    /// # Arguments
+    /// * `index` - the index of array that you what to copy values from
+    /// * `scalar_index` - the index of the scalar value to copy
+    /// * `count` - the number of times to repeat the value
+    ///
+    /// # Panic
+    /// This function panics if there is an invalid index,
+    /// i.e. `index` >= the number of source arrays
+    /// or `start` + `count`  > the length of the `index`th array
+    ///
+    pub fn extend_scalar(&mut self, index: usize, scalar_index: usize, count: usize) {
```

Comment:
```suggestion
    pub fn extend_n(&mut self, index: usize, scalar_index: usize, count: usize) {
```

Scalars don't really exist as a notion at this level